### ISSUE TITLE
✨ Add github action that moves peer review issues  issues as the label on the review is updated across our peer review project board. 

### ISSUE THREAD

**Issue by lwasser** [pyOpenSci/pyosMeta#154]
We currently have a peer review project board [here](https://github.com/orgs/pyOpenSci/projects/7) that we use to track current reviews. this is helpful to keep tabs on our editorial process. Most often editors don't update the review status on the board but they DO update the labels for each issue. 

It easy not too difficult to move an issue to a new "column" or status on the project board using the GitHUB api / graphQL. I have some code that is a proof of concept below. 

There are also actions out there but i'm not sure if any existing actions do what we need them to do. what we want to do is.

1. when an issue has a label change to the label == `0/pre-review-checks` move it to the pre-review-checks status on the board
2. when an issue has a label change to `1-editor-assigned` then move the package to `under-review` status on the board
3. when a issue gets the `6/pyOS-approved 🚀🚀🚀` label, then update the status to `pyos-accepted`
4. when an issue get the `9/joss-approved` label assigned, update the status to `joss-accepted` on the board. 

## I looked into the actions below 
but i haven't been able to get them to work.

* https://github.com/actions/add-to-project
Here is a discussion of doing this using a github action and a github app! 
https://github.com/actions/add-to-project/issues/158

## GitHub action
Once we get the python code working to update the board, we'd next want to create an action that implements this script and updates the board. 
* we could update the board once a week via a cron job. 
* It should also have a workflow dispatch trigger available.  
* it will need to store the token required to update the issue status on the board as a secret. We have an example of doing this with pyos meta in our [contributor update workflow](https://github.com/pyOpenSci/pyopensci.github.io/blob/main/.github/workflows/update-contribs-reviews.yml). 

## Tasks
this issue could be divided into discrete subtasks

- [ ] take the code that i linked to below in the open pr and first clean it up / make it work to update all issues on the board. 
- [ ] Once the code runs and is clean and documented, merge that into pyosmeta
- [ ] Then create a github action that calls the code and updates the project board. this action could live in our [software-submission](https://github.com/pyOpenSci/software-submission)  repo.

it is also fine if we want to refactor my script in smaller chunks during a sprint vs doing all of the work !

**Comment by lwasser** [comment]
@FFY00 here is a gist with my semi working code!

**Comment by FFY00** [comment]
@lwasser, thanks for the code! It was a great help.

I did try to use the REST API, but found that it was broken for newer projects, so I started implementing it with the GraphQL API based on your work, but that quickly lead to something that was more complicated than I think it should be. For that reason, I decided to implement it using github actions directly, which is not as nice, but certainly nicer than the complex script would end up being.

PR: https://github.com/pyOpenSci/software-submission/pull/182

For future reference, here's the code I ended up writing for the script: https://gist.github.com/FFY00/cf322dc62692f25ea321cffca3fa9e77
